lib/server.js

```diff
@@ -20,7 +20,8 @@ app._initializeWSS = function(server) {
   // Create WebSocket server as well.
   this._wss = new WebSocketServer({ path: path, server: server});
 
-  this._wss.on('connection', function(socket) {
+  this._wss.on('connection', function(socket, req) {
+    socket.upgradeReq = req;
     var query = url.parse(socket.upgradeReq.url, true).query;
```

Comment:
If this is the only place where `upgradeReq` is used it is probably better to use:

```js
var query = url.parse(req ? req.url : socket.upgradeReq.url, true).query;
```

and this will work with all `ws` versions.